@xen-orchestra/xapi/host.mjs

```diff
@@ -162,6 +162,24 @@ class Host {
 
     return ipmiSensorsByDataType
   }
+
+  async checkBiosUpdate(ref) {
+    const biosData = await this.call('host.get_bios_strings', ref)
+    const { 'bios-version': currentBiosVersion } = biosData
+
+    const response = await fetch(
+      'https://pictures.2cr.si/Images_site_web_Odoo/Pages_produit/VATES-BIOS_BMC_last-version.json'
+    )
+
+    const parsedData = (await response.json())[0]?.['2CRSi_Servers']
+
+    const serverData = parsedData.find(server => server.Server_Name)
```

Comment:
Your find will always return the first element since your predicate always return `true`.
You need to compare the `server.Server_Name` with the `server_name` of the host